### PR TITLE
fix Send / Sync (typos)

### PR DESCRIPTION
**Pull request by sparecycles**
I think I found a couple copy-paste errors.

### CODE REVIEW

File: src/header/name.rs [hunk 1]
```diff
@@ -2024,7 +2024,7 @@ mod tests {
 
     #[test]
     fn test_bounds() {
-        fn check_bounds<T: Sync + Send>() {}
+        fn check_bounds<T: Send + Sync>() {}
```
Comment (sparecycles) [inline src/header/name.rs]: every other usage is in `Send + Sync` order.

**Comment by carllerche** [comment]
How does HashMap in std implement Send and Sync bounds?

**Comment by carllerche** [comment]
I actually don’t think that Iter can be Send because it holds a reference to the map. If I recall those bounds were intentional.

**Comment by sparecycles** [comment]
Hi @carllerche.  I'm not sure if Iter should be Send or not.

But if that's the case, wouldn't Drain, IterMut, and ValueIterMut have the same bounds?

https://github.com/hyperium/http/blob/8d45bd56007ef341686a54cf092446fe8f988123/src/header/map.rs#L2002-L2003

https://github.com/hyperium/http/blob/8d45bd56007ef341686a54cf092446fe8f988123/src/header/map.rs#L2102-L2103

https://github.com/hyperium/http/blob/8d45bd56007ef341686a54cf092446fe8f988123/src/header/map.rs#L2531-L2532

---

...

https://github.com/hyperium/http/blob/8d45bd56007ef341686a54cf092446fe8f988123/src/header/map.rs#L1941-L1942

**Comment by sparecycles** [comment]
@carllerche you're right! This pattern is also in the rust core libraries.  Still feels like a typo (and if not, that it needs a nice fat comment).

https://github.com/rust-lang/rust/blob/917945d662c42053383fe3e71cb0f313d585e459/src/liballoc/collections/linked_list.rs#L1202-L1218

https://github.com/rust-lang/rust/blob/917945d662c42053383fe3e71cb0f313d585e459/src/libcore/slice/mod.rs#L2650-L2653

https://github.com/rust-lang/rust/blob/master/src/libstd/collections/hash/table.rs#L916-L917

**Comment by sparecycles** [comment]
@carllerche You're right, the pattern of [sync-implies-ref-is-send] is deep in the rust type system.

[sync-implies-ref-is-send]: https://github.com/rust-lang/rust/commit/35ca50bd5676db31a8074a216d1aadad7d434de8

**Comment by carllerche** [comment]
Would you mind adding a comment including your findings for the next person?

**Comment by sparecycles** [comment]
Well, I think I understand why `Iter<T: Sync>` is `Sync + Send`, but I don't understand why `Iter<T: Send>` is not `Send`, nor why `IterMut`/`Drain` etc... don't follow the same pattern.

If a good explanation appears in [rust#53625](https://github.com/rust-lang/rust/issues/53625), I'll can open this with a nice comment.

**Pull request closed without merging**